test/net/sourceforge/kolmafia/persistence/ItemFinderTest.java

```diff
@@ -906,20 +902,68 @@ public void itShouldParseMultipleItems() {
   @ParameterizedTest
   @CsvSource({
     "'1 1337 7r0uZ0RZ'," + ItemPool.ELITE_TROUSERS + ",1",
+    // space in name that begins with digit, 1 assumed for quantity
     "'1337 7r0uZ0RZ'," + ItemPool.ELITE_TROUSERS + ",1",
+    // See https://wiki.kolmafia.us/index.php/CLI_Reference 1 WA fuzzy matched to 100-watt light
+    // bulb
     "'1 WA'," + ItemPool.WA + ",1",
     "'1 Meat'," + ItemPool.ONE_MEAT + ",1",
     "'1 1 Meat'," + ItemPool.ONE_MEAT + ",1",
     "'123 1 Meat'," + ItemPool.ONE_MEAT + ",123",
+    // Initial bug was 1 7-ball being returned as 17-ball
     "'1 7-ball'," + ItemPool.SEVEN_BALL + ",1"
   })
   public void parseStringAndFindItemAndQuantity(
       String toBeParsed, int expectedItemId, int expectedQuantity) {
     AdventureResult item;
     item = ItemFinder.getFirstMatchingItem(toBeParsed, false, null, Match.ANY);
     assertEquals(StaticEntity.getContinuationState(), MafiaState.CONTINUE);
-    assertTrue(item != null);
+    assertNotNull(item);
     assertEquals(expectedItemId, item.getItemId());
     assertEquals(expectedQuantity, item.getCount());
   }
+
+  // Tests written for the sole purpose of 100% coverage
+  @Test
+  public void itShouldMatchANYWhenNotExplicitlySpecified() {
+    List<String> nameList = new ArrayList<>();
+    String item;
+    nameList.add("7-ball");
+    nameList.add("17-ball");
+    item = ItemFinder.getFirstMatchingItemName(nameList, "7-ball");
+    assertEquals(StaticEntity.getContinuationState(), MafiaState.CONTINUE);
+    assertNotNull(item);
+    assertEquals(item, "7-ball");
+    nameList.clear();
+    nameList.add("17-ball");
+    nameList.add("7-ball");
+    item = ItemFinder.getFirstMatchingItemName(nameList, "7-ball");
+    assertEquals(StaticEntity.getContinuationState(), MafiaState.CONTINUE);
+    assertNotNull(item);
+    assertEquals(item, "7-ball");
+  }
+
+  @Test
+  public void itShouldReturnNullWhenPassedNullOrEmptyList() {
+    List<String> nameList = new ArrayList<>();
+    String item;
+    item = ItemFinder.getFirstMatchingItemName(null, "7-ball");
```

Comment:
Again, this could be parameterised rather than repeating a test interally (and thus obfuscating later failures)